### PR TITLE
montrer-le-bureau@cannelle.org: Add l10n support

### PR DESCRIPTION
**Pull request by cranes-bill**
[REf](https://github.com/linuxmint/cinnamon-spices-applets/pull/492)

### CODE REVIEW

File: montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/applet.js [hunk 1]
```diff
@@ -36,232 +33,232 @@ MyApplet.prototype = {
...
+
+        let CMDLINE = "vlc screen://  --qt-start-minimized --input-slave alsa:// --no-video " + USELOGO + "--screen-width=" + SCRW + " --screen-height=" + SCRH + " :screen-fps=" + VFPS + " :screen-caching=" + SCASH + " ";
+
+        let TRANSCODING = "--sout " + "\"#transcode{vcodec=" + VCOD + VFCROP + ",vb=" + VBR + ",fps=" + VFPS + ",scale=" + "1" + ",acodec=" + ACOD + ",ab=" + ABR + ",channels=" + ACHAN + ",samplerate=" + SMPLRT + LOGOFLTR + "}:duplicate{dst=std{access=file,mux=" + MUXR + ",dst=" + DIRNAME + "/" + ONAME + N_O_W + MUXR + "}}\"";
+
+        let lock_file = Gio.file_new_for_path(LOCKFILE);
+        if (lock_file.query_exists(null)) {
+            new ModalDialog.NotifyDialog(_("Sorry, only one instance allowed!") + "\n").open();
+        } else if ((this.CROP_L >= SCRW) || (this.CROP_R >= SCRW) || (this.CROP_B >= SCRH) || (this.CROP_T >= SCRH)) {
+            new ModalDialog.NotifyDialog(_("Sorry, the record can't starting,") + "\n" + _("because you have incorrect crop settings!") + "\n").open();
```
Comment (giwhub) [inline montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/applet.js]: Sorry, the record can't start.

File: montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/applet.js [hunk 2]
```diff
@@ -36,232 +33,232 @@ MyApplet.prototype = {
...
-                new ModalDialog.NotifyDialog(NORECMESSAGE).open();
-                }
+        let lock_file = Gio.file_new_for_path(LOCKFILE);
+        if (lock_file.query_exists(null)) {
+            Util.spawnCommandLine("killall -SIGTERM vlc");
+            Util.spawnCommandLine("rm -f " + LOCKFILE);
+            Util.spawnCommandLine("notify-send --icon=process-stop \"Recording has finished\"");
+            this.set_applet_icon_name(DefIcon);
+        } else {
+            new ModalDialog.NotifyDialog(_("Ther is no active recording.") + "\n").open();
```
Comment (giwhub) [inline montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/applet.js]: There is no active recording.

File: montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/settings-schema.json [hunk 3]
```diff
@@ -271,15 +271,15 @@
...
+        "tooltip": "Adding ability to take some action with mouse middle button"
 },
     "ss-tool": {
         "type": "filechooser",
         "dependency": "use-screen-shot",
-        "description": "Screenshot tool:",
+        "description": "Middle button tool:",
         "default": "/usr/bin/gnome-screenshot",
-        "tooltip": "Select the full path to screenshot tool. If you want to pass some arguments, you might create simple shell script, then select it here."
+        "tooltip": "Select the full path to middle button tool. If you want to pass some arguments, you might create simple shell script, then select it here."
```
Comment (giwhub) [inline montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/settings-schema.json]: you might create a simple shell script

**Comment by cranes-bill** [comment]
Thanks!

**Comment by cranes-bill** [comment]
Should i change something other, or just to close it if it's no good to be merged!?

**Comment by NikoKrause** [comment]
Is this ready to merge?

Suggestions:
1.  `_("Sorry, the record can't start,") + "\n" + _("because you have incorrect crop settings!")`
This is one sentence. I wouldn't split it into two translatable Strings.
`_("Sorry, the record can't start,\n because you have incorrect crop settings!")`
2. If you want to create a template for translations, then you need a folder called `po`, where the metadata.json file is stored.
Then open a terminal. Direct to the place oft you applet, where the metadata is störend: `cd ~/.local/share/cinnamon/applets/YOURAPPLETUUID/`
Then the command to create the template. It's also used to update the template:
`cinnamon-json-makepot --js po/YOURAPPLERUUID.pot`

**Comment by cranes-bill** [comment]
> This is one sentence. I wouldn't split it into two translatable Strings.

Ooops, i was not mentioned that, sorry!

> If you want to create a template for translations, then you need a folder called po, where the metadata.json file is stored.

I missed that step, because @giwhub had already created one in his [Pull request](https://github.com/linuxmint/cinnamon-spices-applets/pull/492)
Sorry!

**Comment by cranes-bill** [comment]
I think it's ready now.

File: montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/po/montrer-le-bureau@cannelle.org.pot [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+msgstr ""
+
+#. montrer-le-bureau@cannelle.org->settings-schema.json->crop-
+#. left->description
+msgid "Crop from the left:"
+msgstr ""
+
+#. montrer-le-bureau@cannelle.org->settings-schema.json->crop-left->tooltip
+#. montrer-le-bureau@cannelle.org->settings-schema.json->crop-right->tooltip
+msgid "The value must be lower then your screen width"
```
Comment (giwhub) [inline montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/po/montrer-le-bureau@cannelle.org.pot]: The value must be lower than your screen width

File: montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/po/montrer-le-bureau@cannelle.org.pot [hunk 5]
```diff
@@ -0,0 +1,492 @@
...
+msgid "Show or record desktop"
+msgstr ""
+
+#. montrer-le-bureau@cannelle.org->settings-schema.json->crop-top->description
+msgid "Crop from the top:"
+msgstr ""
+
+#. montrer-le-bureau@cannelle.org->settings-schema.json->crop-top->tooltip
+#. montrer-le-bureau@cannelle.org->settings-schema.json->crop-bottom->tooltip
+msgid "The value must be lower then your screen height"
```
Comment (giwhub) [inline montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/po/montrer-le-bureau@cannelle.org.pot]: The value must be lower than your screen height

File: montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/po/montrer-le-bureau@cannelle.org.pot [hunk 6]
```diff
@@ -0,0 +1,492 @@
...
+msgid "Full path of the image file to use."
+msgstr ""
+
+#. montrer-le-bureau@cannelle.org->settings-schema.json->file-
+#. name->description
+msgid "Output file name:"
+msgstr ""
+
+#. montrer-le-bureau@cannelle.org->settings-schema.json->file-name->tooltip
+msgid "Just a name. The time and extension will be add automatically."
```
Comment (giwhub) [inline montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/po/montrer-le-bureau@cannelle.org.pot]: The time and extension will be added automatically.

File: montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/po/montrer-le-bureau@cannelle.org.pot [hunk 7]
```diff
@@ -0,0 +1,492 @@
...
+msgstr ""
+
+#. montrer-le-bureau@cannelle.org->settings-schema.json->use-screen-
+#. shot->tooltip
+msgid "Adding ability to take some action with mouse middle button"
+msgstr ""
+
+#. montrer-le-bureau@cannelle.org->settings-schema.json->video-filter-
+#. crop->description
+msgid "Check this to select a cropping areas:"
```
Comment (giwhub) [inline montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/po/montrer-le-bureau@cannelle.org.pot]: Check this to select a cropping area:

File: montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/po/montrer-le-bureau@cannelle.org.pot [hunk 8]
```diff
@@ -0,0 +1,492 @@
...
+msgstr ""
+
+#. montrer-le-bureau@cannelle.org->settings-schema.json->video-filter-
+#. crop->tooltip
+msgid ""
+"Checking this box allows you to set a custom areas from the screen, that "
```
Comment (giwhub) [inline montrer-le-bureau@cannelle.org/files/montrer-le-bureau@cannelle.org/po/montrer-le-bureau@cannelle.org.pot]: Checking this box allows you to set a custom area from the screen

**Comment by cranes-bill** [comment]
That's what i meant when i said i'm not sure the strings are correct. Initially i have been creating this applet just for myself, just to add an additional function to the original applet, and kill me if i know why i decide to share it. Anyway..
Sorry, and thank you very much!

**Comment by giwhub** [comment]
You're welcome,no worries,buddy.
Not a big problem actually.

**Pull request merged**